api/v3/indexercluster_types.go

```diff
@@ -134,3 +140,34 @@ type IndexerClusterList struct {
 func init() {
 	SchemeBuilder.Register(&IndexerCluster{}, &IndexerClusterList{})
 }
+
+// NewEvent creates a new event associated with the object and ready
+// to be published to the kubernetes API.
+func (icstr *IndexerCluster) NewEvent(eventType, reason, message string) corev1.Event {
+	t := metav1.Now()
+	return corev1.Event{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: reason + "-",
+			Namespace:    icstr.ObjectMeta.Namespace,
+		},
+		InvolvedObject: corev1.ObjectReference{
+			Kind:       "IndexCluster",
+			Namespace:  icstr.Namespace,
+			Name:       icstr.Name,
+			UID:        icstr.UID,
+			APIVersion: GroupVersion.String(),
+		},
+		Reason:  reason,
+		Message: message,
+		Source: corev1.EventSource{
+			Component: "splunk-indexcluster-controller",
+		},
+		FirstTimestamp:      t,
+		LastTimestamp:       t,
+		Count:               1,
+		Type:                eventType,
+		ReportingController: "enterprise.splunk.com/indexcluster-controller",
+		//Related:             standln.Spec.ConsumerRef,
+		//Action: TODO FIXME what action can be taken
+	}
+}
```

Comment:
I thought about it, but I felt it looks ugly. in future if we want to do specific actions to these events we can add it here. since its only 5 CR at this moment, we can keep the way it is. if we are adding more we can consolidate all of them. let me know what you think

Reply:
generics will be introduced to golang in next release. we can write generic single function at that time.

Reply:
removed